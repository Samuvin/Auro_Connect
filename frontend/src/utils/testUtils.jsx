import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

// Create a custom render function that includes providers
export const renderWithProviders = (ui, options = {}) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  });

  const AllTheProviders = ({ children }) => {
    return (
      <QueryClientProvider client={queryClient}>
        <BrowserRouter>
          {children}
        </BrowserRouter>
      </QueryClientProvider>
    );
  };

  return render(ui, { wrapper: AllTheProviders, ...options });
};

// Mock user data for testing
export const mockUser = {
  _id: '1',
  name: 'John Doe',
  username: 'johndoe',
  email: 'john@example.com',
  profilePicture: '/avatar.png',
  headline: 'Software Developer',
  connections: ['2', '3'],
  location: 'New York, NY'
};

// Mock post data for testing
export const mockPost = {
  _id: '1',
  content: 'This is a test post',
  author: mockUser,
  likes: ['2', '3'],
  comments: [
    {
      _id: '1',
      content: 'Great post!',
      user: mockUser,
      createdAt: new Date().toISOString()
    }
  ],
  image: 'https://example.com/image.jpg',
  createdAt: new Date().toISOString()
}; 