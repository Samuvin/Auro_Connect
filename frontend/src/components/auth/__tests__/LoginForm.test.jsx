import { screen, fireEvent, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import LoginForm from '../LoginForm';
import { renderWithProviders } from '../../../utils/testUtils';
import * as reactQuery from '@tanstack/react-query';

// Mock the dependencies
jest.mock('../../../lib/axios');

describe('LoginForm', () => {
  const mockMutate = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
    
    jest.spyOn(reactQuery, 'useMutation').mockReturnValue({
      mutate: mockMutate,
      isLoading: false,
      error: null
    });
  });

  test('renders form correctly', () => {
    renderWithProviders(<LoginForm />);
    
    expect(screen.getByPlaceholderText('Username')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Password')).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /login/i })).toBeInTheDocument();
  });

  test('form fields are required', () => {
    renderWithProviders(<LoginForm />);
    
    const usernameInput = screen.getByPlaceholderText('Username');
    const passwordInput = screen.getByPlaceholderText('Password');
    
    expect(usernameInput).toBeRequired();
    expect(passwordInput).toBeRequired();
  });

  test('password field has correct type', () => {
    renderWithProviders(<LoginForm />);
    
    const passwordInput = screen.getByPlaceholderText('Password');
    expect(passwordInput).toHaveAttribute('type', 'password');
  });

  test('form accepts user input', async () => {
    const user = userEvent.setup();
    renderWithProviders(<LoginForm />);
    
    const usernameInput = screen.getByPlaceholderText('Username');
    const passwordInput = screen.getByPlaceholderText('Password');
    
    await user.type(usernameInput, 'testuser');
    await user.type(passwordInput, 'password123');
    
    expect(usernameInput).toHaveValue('testuser');
    expect(passwordInput).toHaveValue('password123');
  });

  // Snapshot Tests
  test('matches snapshot for login form default state', () => {
    const { container } = renderWithProviders(<LoginForm />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('matches snapshot for login form loading state', () => {
    jest.spyOn(reactQuery, 'useMutation').mockReturnValue({
      mutate: mockMutate,
      isLoading: true,
      error: null
    });

    const { container } = renderWithProviders(<LoginForm />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('matches snapshot for login form with error state', () => {
    jest.spyOn(reactQuery, 'useMutation').mockReturnValue({
      mutate: mockMutate,
      isLoading: false,
      error: { message: 'Invalid credentials' }
    });

    const { container } = renderWithProviders(<LoginForm />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('matches snapshot for login form with filled inputs', async () => {
    const user = userEvent.setup();
    const { container } = renderWithProviders(<LoginForm />);
    
    const usernameInput = screen.getByPlaceholderText('Username');
    const passwordInput = screen.getByPlaceholderText('Password');
    
    await user.type(usernameInput, 'testuser');
    await user.type(passwordInput, 'password123');
    
    expect(container.firstChild).toMatchSnapshot();
  });
}); 