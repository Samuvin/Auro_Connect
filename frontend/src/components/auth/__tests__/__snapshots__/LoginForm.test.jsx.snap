// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`LoginForm matches snapshot for login form default state 1`] = `
<form
  class="space-y-4 w-full max-w-md"
>
  <input
    class="input input-bordered w-full"
    placeholder="Username"
    required=""
    type="text"
    value=""
  />
  <input
    class="input input-bordered w-full"
    placeholder="Password"
    required=""
    type="password"
    value=""
  />
  <button
    class="btn btn-primary w-full"
    type="submit"
  >
    Login
  </button>
</form>
`;

exports[`LoginForm matches snapshot for login form loading state 1`] = `
<form
  class="space-y-4 w-full max-w-md"
>
  <input
    class="input input-bordered w-full"
    placeholder="Username"
    required=""
    type="text"
    value=""
  />
  <input
    class="input input-bordered w-full"
    placeholder="Password"
    required=""
    type="password"
    value=""
  />
  <button
    class="btn btn-primary w-full"
    type="submit"
  >
    Login
  </button>
</form>
`;

exports[`LoginForm matches snapshot for login form with error state 1`] = `
<form
  class="space-y-4 w-full max-w-md"
>
  <input
    class="input input-bordered w-full"
    placeholder="Username"
    required=""
    type="text"
    value=""
  />
  <input
    class="input input-bordered w-full"
    placeholder="Password"
    required=""
    type="password"
    value=""
  />
  <button
    class="btn btn-primary w-full"
    type="submit"
  >
    Login
  </button>
</form>
`;

exports[`LoginForm matches snapshot for login form with filled inputs 1`] = `
<form
  class="space-y-4 w-full max-w-md"
>
  <input
    class="input input-bordered w-full"
    placeholder="Username"
    required=""
    type="text"
    value="testuser"
  />
  <input
    class="input input-bordered w-full"
    placeholder="Password"
    required=""
    type="password"
    value="password123"
  />
  <button
    class="btn btn-primary w-full"
    type="submit"
  >
    Login
  </button>
</form>
`;
